framework/wazuh/cluster/dapi/dapi.py

```diff
@@ -218,9 +218,23 @@ def run_local(args):
         # get the node(s) who has all available information to answer the request.
         nodes = self.get_solver_node()
         self.input_json['from_cluster'] = True
+        # apply offset parameter when merging results from nodes
+        if 'offset' in self.input_json['arguments']:
+            offset = self.input_json['arguments']['offset']  # save offset
+            self.input_json['arguments']['offset'] = 0  # set offset=0 before distributing call

```

Comment:
I don't agree with this solution. I'm not sure it solves the problem. Actually I think it breaks the offset for the rest of endpoints. 

We cannot risk the correctness of common endpoints just to fix the experimental ones. We should think another approach and then come back with a better solution.

Reply:
`forward_request` method is called only on `distributed_master` requests.

We have the follow endpoints of `distributed_master` type:

- `GET /cluster/:node_id/logs`
- `GET /rootcheck/:agent_id/cis`
- `GET /rootcheck/:agent_id`
- `GET /rootcheck/:agent_id/pci`
- `GET /sca/:agent_id/checks/:id`
- `GET /sca/:agent_id`
- `GET /syscollector/:agent_id/packages`
- `GET /syscollector/:agent_id/processes`
- `GET /syscollector/:agent_id/ports`
- `GET /syscollector/:agent_id/netaddr`
- `GET /syscollector/:agent_id/netiface`
- `GET /syscollector/:agent_id/netproto`
- `GET /ciscat/:agent_id/results`
- `GET /experimental/syscollector/os`
- `GET /experimental/syscollector/hardware`
- `GET /experimental/syscollector/packages`
- `GET /experimental/syscollector/processes`
- `GET /experimental/syscollector/ports`
- `GET /experimental/syscollector/netaddr`
- `GET /experimental/syscollector/netproto`
- `GET /experimental/syscollector/netiface`
- `GET /experimental/ciscat/results`

Only `experimentals` calls merge results, so I think I should move the changes inside this conditional in order to avoid secondary effects and fix pagination in `experimental` endpoints:
https://github.com/wazuh/wazuh/blob/939481f3dc06294ba7150dfb569430dcf6d76f3d/framework/wazuh/cluster/dapi/dapi.py#L226

Thoughts?